static_frame/test/unit/test_batch.py

```diff
@@ -826,7 +827,325 @@ def test_batch_roll_a(self) -> None:
                 )
 
     #---------------------------------------------------------------------------
+    def test_batch_isna(self) -> None: # also tests `notna()`
+        f0 = ff.parse('v(str,str,bool,float)|s(9,4)').assign[3]([None, 100.0, 632.23, None, 12.5, 51526.002, None, None, 0.231])
+        actual1 = Batch.from_frames((
+            f0.iloc[0:3].rename('1'),
+            f0.iloc[3:6].rename('2'),
+            f0.iloc[6:9].rename('3'),
+        )).isna().to_frame()[3].values
+        actual2 = Batch.from_frames((
+            f0.iloc[0:3].rename('1'),
+            f0.iloc[3:6].rename('2'),
+            f0.iloc[6:9].rename('3'),
+        )).notna().to_frame()[3].values
+        actual2 = np.invert(actual2)
+
+        expected = np.array([True, False, False, True, False, False, True, True, False])
+        self.assertTrue(((actual1 == actual2) & (actual2 == expected)).all())
```

Comment:
Here and elsewhere, better to break out this comparison to compare one array at a time, and better to use something like `self.assertEquals(actual1.tolist(), actual2.tolist())`, as, on failure, it will provide better visibility into what is wrong.